kernel/mutex_sleep/src/mutex.rs

```diff
@@ -3,14 +3,15 @@ use core::ops::{Deref, DerefMut};
 use spin::{Mutex, MutexGuard};
 use wait_queue::WaitQueue;
 use lockable::{Lockable, LockableSized};
+use sync_spin::Spin;
 
 /// A mutual exclusion wrapper that puts a `Task` to sleep while waiting for the lock to become available. 
 /// 
 /// A sleeping `Task` has a "blocked" runstate, meaning that it will not be scheduled in. 
 /// Once the lock becomes available, `Task`s that are sleeping while waiting for the lock
 /// will be notified (woken up) so they can attempt to acquire the lock again.
 pub struct MutexSleep<T: ?Sized> {
-    queue: WaitQueue,
+    queue: WaitQueue<Spin>,
```

Comment:
as in my other comment, i don't think it's needed to specify `<Spin>` here since it's the default (?)